src/components/SigningModal/SignMessageFlow.js

```diff
@@ -127,9 +136,14 @@ function SignMessageFlow({ signatureBag, visible, apps, onClose }) {
           </div>
         ),
       },
-    ],
-    [message, requestingApp, account, infoDescriptions, handleMsgSign]
-  )
+    ]
+
+    if (walletError) {
```

Comment:
Ah, in terms of these `if`s, does it make sense to lift them as high up as possible? Since you don't need to know about how `signMessage` is constructed or used if you have a `walletError`.

Reply:
Yeah that's a good point.